### DISCUSSION TITLE
Can I use a virtual or cloud Mac for IOS development?

### DISCUSSION THREAD

**Discussion by otium99** [dotnet/maui#7372] [Q&A] [answered]
Hello,

I am just discovering .NET Maui.

In this course https://www.youtube.com/watch?v=0PBVSMToA3A he talks about, that you need a Mac for IOS development. I don't have a Mac. What are the best alternative options?

- Deploying something like this https://www.wikihow.com/Install-Macos-on-a-Virtual-Machine to either my local machine or a Windows Server
- Using a service like this: https://www.macincloud.com/
- What else?

Does it even work with a Mac OS in a virtual machine?

Thanks

**Comment by chezchez** [comment]
I’m DYING TO KNOW also! Thx

**Comment by kcrg** [comment]
I test option with virtual machine but that was really painful and slow (eg installing Xcode took I believe whole day) even on solid PC (R9 5900x and nvme ssd).
I think the best option is to just buy Mac, sadly.

**Comment by otium99** [comment, marked as answer]
I am working now with https://www.macincloud.com and it works. The simulator takes about 15 minutes to load! But at least I was able to create a running *.ipa file and transfer it to my phone through ad-hoc distribution and a third party service https://www.installonair.com/

With this setting I can work now. Develop on the Windows machine as a windows app and then distribute to the simulator/physical phone.

**Reply by sjbthfc2** [reply]
How did you get this to work, I have a Mac instance on AWS EC2 and I can SSH into it from the Windows machine but when pairing from Visual Studio on Windows I get "Couldn't resolve Mac".

Thanks

**Comment by xtuzy** [comment]
I use Hackintosh

**Comment by SamuelBachar** [comment]
Is it possible to also sing .net maui application and release / publish application to AppStore using only https://www.macincloud.com/ ?

**Reply by JohnHDev** [reply]
Probably, but your better option might be to use a service like Azure DevOps, we use it for virtual build hosts and I use it personally for git as well as builds. Can't use it for debugging though, but its great for CICD, code repos in git, work items etc.
